morpheus/controllers/elasticsearch_controller.py

```diff
@@ -35,22 +36,21 @@ class ElasticsearchController:
         Whether to raise exceptions on Elasticsearch errors.
     refresh_period_secs : int, optional, default: 2400
         The refresh period in seconds for client refreshing.
-    pickled_func_config : dict, optional, default: None
-        Configuration for a pickled function to modify connection parameters.
+    connection_kwargs_update_func : typing.Callable, optional, default: None
+        Custom function to update connection parameters.
     """
 
     def __init__(self,
                  connection_kwargs: dict,
                  raise_on_exception: bool = False,
                  refresh_period_secs: int = 2400,
-                 pickled_func_config: dict = None):
+                 connection_kwargs_update_func: typing.Callable = None):
 
         self._client = None
         self._last_refresh_time = None
-        self._connection_kwargs = connection_kwargs
         self._raise_on_exception = raise_on_exception
         self._refresh_period_secs = refresh_period_secs
-        self._apply_derive_params_func(pickled_func_config)
+        self._connection_kwargs = self._apply_custom_func(connection_kwargs_update_func, connection_kwargs)
```

Comment:
I wasn't sure how to test the custom function when it's being used in the `write_to_elasticsearch.py` module because it relies on the Elasticsearch server. So, I opted that approach.